pkg/plugins/runtime/k8s/plugin.go

```diff
@@ -125,8 +135,65 @@ func addValidators(mgr kube_ctrl.Manager, rt core_runtime.Runtime) error {
 	mgr.GetWebhookServer().Register(path, composite.WebHook())
 	log.Info("Registering a validation composite webhook", "path", path)
 
-	mgr.GetWebhookServer().Register("/validate-v1-service", &kuba_webhook.Admission{Handler: &k8s_webhooks.ServiceValidator{}})
+	mgr.GetWebhookServer().Register("/validate-v1-service", &kube_webhook.Admission{Handler: &k8s_webhooks.ServiceValidator{}})
 	log.Info("Registering a validation webhook for v1/Service", "path", "/validate-v1-service")
 
 	return nil
 }
+
+func addMutators(mgr kube_ctrl.Manager, rt core_runtime.Runtime) {
+	mgr.GetWebhookServer().Register("/inject-sidecar",
+		PodMutatingWebhook(injector.New(rt.Config().Runtime.Kubernetes.Injector, mgr.GetClient()).InjectKuma))
+}
+
+//func Setup(mgr kube_manager.Manager, cfg *kuma_injector_conf.Config) error {
+//	webhookServer := &kube_webhook.Server{
+//		Host:    cfg.WebHookServer.Address,
+//		Port:    int(cfg.WebHookServer.Port),
+//		CertDir: cfg.WebHookServer.CertDir,
+//	}
+//	if err := mesh_k8s.AddToScheme(mgr.GetScheme()); err != nil {
+//		return errors.Wrap(err, "could not add to scheme")
+//	}
+//
+//	if err := k8scnicncfio.AddToScheme(mgr.GetScheme()); err != nil {
+//		return errors.Wrap(err, "could not add to scheme")
+//	}
+//
+//	webhookServer.Register("/inject-sidecar", PodMutatingWebhook(injector.New(cfg.Injector, mgr.GetClient()).InjectKuma))
+//	webhookServer.WebhookMux.HandleFunc("/healthy", func(resp http.ResponseWriter, _ *http.Request) {
+//		resp.WriteHeader(http.StatusOK)
+//	})
+//	webhookServer.WebhookMux.HandleFunc("/ready", func(resp http.ResponseWriter, _ *http.Request) {
+//		resp.WriteHeader(http.StatusOK)
+//	})
+//	return mgr.Add(webhookServer)
+//}
+
+type PodMutator func(*kube_core.Pod) error
```

Comment:
How about moving PodMutator to injector package?

Reply:
Maybe better to place it to `webhook` package? Looks like its logic might be easy reused

Reply:
ok, webhook is also ok. I'd like to move it out of the plugin.go if possible